LEAN-3978 LW-FY25-Q2-S3: Table_element delete suggestion is not showi…

fwaisi opened the pull request:
…… #427

fwaisi committed:
LEAN-3978 LW-FY25-Q2-S3: Table_element delete suggestion is not showi… #427

fwaisi committed:
Revert "LEAN-3978 LW-FY25-Q2-S3: Table_element delete suggestion is not showi… #427"

This reverts commit c02ec91d05b6ce42b7b979189ca03f525a00069f.

zax-assistant[bot] committed:
Update package.json

fwaisi committed:
LEAN-3978 LW-FY25-Q2-S3: Table_element delete suggestion is not showi… …

fwaisi commented:
Two problems were identified and fixed in the track changes plugin:

- Prosemirror was attempting to insert a table node inside a paragraph node, which is invalid in the document schema. This was causing an error.

- The splitSliceIntoMergedParts function was inadvertently removing paragraph nodes from the slice because it was attempting to merge them with incompatible nodes (like tables). To fix this, I added a node compatibility check before attempting any merge operations:

`const firstNodeType = firstChild?.type?.name
const nextNodeType = nodes[1]?.type?.name
// Only merge when nodes are of the same type
const firstMergedNode = openStart > 0 && mergeSides && firstChild && firstNodeType === nextNodeType`

I also fixed a logic error in [this condition](https://github.com/Atypon-OpenSource/manuscripts-track-changes-plugin/blob/91ba8fff1bdeefb152e3169f67cd037f81d33351/src/mutate/deleteAndMergeSplitNodes.ts#L196) which was incorrectly evaluating to true in all cases.

IslamJMomani commented:
> Two problems were identified and fixed in the track changes plugin:
> 
> * Prosemirror was attempting to insert a table node inside a paragraph node, which is invalid in the document schema. This was causing an error.
> * The splitSliceIntoMergedParts function was inadvertently removing paragraph nodes from the slice because it was attempting to merge them with incompatible nodes (like tables). To fix this, I added a node compatibility check before attempting any merge operations:
> 
> `const firstNodeType = firstChild?.type?.name const nextNodeType = nodes[1]?.type?.name // Only merge when nodes are of the same type const firstMergedNode = openStart > 0 && mergeSides && firstChild && firstNodeType === nextNodeType`
> 
> I also fixed a logic error in [this condition](https://github.com/Atypon-OpenSource/manuscripts-track-changes-plugin/blob/91ba8fff1bdeefb152e3169f67cd037f81d33351/src/mutate/deleteAndMergeSplitNodes.ts#L196) which was incorrectly evaluating to true in all cases.

Are these problems occurring only when pasting the table node, rather than the entire table element? In other words, if the selection includes the whole table element, do any issues still arise?

I’m still under the impression that this is a selection issue that could be handled in the body editor. I’m inviting @mbartenev-atypon  to take a look.

fwaisi commented:
> > Two problems were identified and fixed in the track changes plugin:
> > 
> > * Prosemirror was attempting to insert a table node inside a paragraph node, which is invalid in the document schema. This was causing an error.
> > * The splitSliceIntoMergedParts function was inadvertently removing paragraph nodes from the slice because it was attempting to merge them with incompatible nodes (like tables). To fix this, I added a node compatibility check before attempting any merge operations:
> > 
> > `const firstNodeType = firstChild?.type?.name const nextNodeType = nodes[1]?.type?.name // Only merge when nodes are of the same type const firstMergedNode = openStart > 0 && mergeSides && firstChild && firstNodeType === nextNodeType`
> > I also fixed a logic error in [this condition](https://github.com/Atypon-OpenSource/manuscripts-track-changes-plugin/blob/91ba8fff1bdeefb152e3169f67cd037f81d33351/src/mutate/deleteAndMergeSplitNodes.ts#L196) which was incorrectly evaluating to true in all cases.
> 
> Are these problems occurring only when pasting the table node, rather than the entire table element? In other words, if the selection includes the whole table element, do any issues still arise?
> 
> I’m still under the impression that this is a selection issue that could be handled in the body editor. I’m inviting @mbartenev-atypon to take a look.

@IslamJMomani, yes, it's occurring when pasting the table node. Updating the selection as I did in the beginning will fix the issue, but disabling the track-changes plugin will fix the issue as well. So I think we need to decide whether to fix it in the track-changes or in the body-editor.

@mbartenev-atypon, please advise?
